pkg/peertls/peertls.go

```diff
@@ -110,6 +117,28 @@ func VerifyPeerCertChains(_ [][]byte, parsedChains [][]*x509.Certificate) error
 	return verifyChainSignatures(parsedChains[0])
 }
 
+// VerifyCAWhitelist verifies that the peer identity's leaf was signed by any one of the
+// (certificate authority) certificates in the provided whitelist
+func VerifyCAWhitelist(cas []*x509.Certificate) PeerCertVerificationFunc {
+	if cas == nil {
+		return nil
+	}
+
+	return func(_ [][]byte, parsedChains [][]*x509.Certificate) error {
+		var (
+			err   error
```

Comment:
might want to use that pkg/utils CombineErrors()

Reply:
nice - I didn't know this was a thing.